cdap-security/src/test/java/co/cask/cdap/security/authorization/InMemoryAuthorizer.java

```diff
@@ -69,9 +67,9 @@ public void enforce(EntityId entity, Principal principal, Set<Action> actions) t
     if (superUsers.contains(principal) || superUsers.contains(allSuperUsers)) {
       return;
     }
-    // actions allowed to this principal
+    // actions allowed for this principal
     Set<Action> allowed = Sets.newHashSet(getActions(entity, principal));
-    // actions allowed to any of the roles to which this principal belongs if its not a role
+    // actions allowed for any of the roles to which this principal belongs if its not a role
     if (principal.getType() != Principal.PrincipalType.ROLE) {
       for (Role role : listRoles(principal)) {
```

Comment:
call `listRoles` has unnecessary overhead of copying to a new set. Why not just loop over the `roleToPrincipals` directly?